BugsnagNetworkRequestPlugin/BugsnagNetworkRequestPlugin/BSGURLSessionTracingDelegate.m

```diff
@@ -72,32 +79,24 @@ - (void)URLSession:(NSURLSession *)session
         NSURLComponents *urlComponents = [NSURLComponents componentsWithURL:req.URL resolvingAgainstBaseURL:YES];
         // Note: Cannot use metrics transaction response because it will be nil if a custom NSURLProtocol is present.
         // Note: If there was an error, task.response will be nil, and the following values will be set accordingly.
-        NSURLResponse *resp = task.response;
-        NSHTTPURLResponse *httpResp = [resp isKindOfClass:NSHTTPURLResponse.class] ? (NSHTTPURLResponse *)resp : nil;
-        NSString *message = [self conclusionForResponseCode:httpResp.statusCode];
-        int64_t requestContentLength = (int64_t)req.HTTPBody.length;
-        int64_t responseContentLength = resp.expectedContentLength;
-
-        if (@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)) {
-            NSURLSessionTaskTransactionMetrics *transaction = [metrics.transactionMetrics lastObject];
-            // Note: Must check for zero because these somtimes lie.
-            if (transaction.countOfRequestBodyBytesSent != 0) {
-                requestContentLength= transaction.countOfRequestBodyBytesSent;
-            }
-            if (transaction.countOfResponseBodyBytesReceived != 0) {
-                responseContentLength = transaction.countOfResponseBodyBytesReceived;
-            }
+        NSHTTPURLResponse *httpResp = [task.response isKindOfClass:NSHTTPURLResponse.class] ? (NSHTTPURLResponse *)task.response : nil;
+
+        NSString *message = [BSGURLSessionTracingDelegate messageForResponse:httpResp];
+
+        NSMutableDictionary *metadata = [NSMutableDictionary dictionary];
+        metadata[@"duration"] = @((unsigned)(metrics.taskInterval.duration * 1000));
+        metadata[@"method"] = req.HTTPMethod;
+        metadata[@"url"] = [BSGURLSessionTracingDelegate URLStringWithoutQueryForComponents:urlComponents];
+        metadata[@"urlParams"] = [BSGURLSessionTracingDelegate urlParamsForQueryItems:urlComponents.queryItems];
+        if (req.HTTPBody) {
+            metadata[@"requestContentLength"] = @(req.HTTPBody.length);
```

Comment:
Why httpbody.length and not `task.countOfBytesSent`?

Reply:
`task.countOfBytesSent` returns erroneous values when an NSURLProtocol is in use.

Reply:
`task.countOfBytesSent` is `0` when a custom NSURLProtocol is used, which breaks our unit tests.

Reply:
But `HTTPBody.length` is broken because there isn't always a body - `HTTPBodyStream` may have been set instead.

Reply:
Now using `task.countOfBytesSent` when available